rpc/client.go

```diff
@@ -0,0 +1,90 @@
+package rpc // import "collectd.org/rpc"
+
+import (
+	"io"
+	"log"
+
+	"collectd.org/api"
+	pb "collectd.org/rpc/proto"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+)
+
+// client is a wrapper around pb.CollectdClient implementing Interface.
+type client struct {
+	ctx    context.Context
+	client pb.CollectdClient
+}
+
+// Newclient returns a wrapper around the gRPC client connection that maps
+// between the Go interface and the gRPC interface.
+func NewClient(ctx context.Context, conn *grpc.ClientConn) Interface {
+	return &client{
+		ctx:    ctx,
+		client: pb.NewCollectdClient(conn),
+	}
+}
+
+// Query maps its arguments to a QueryValuesRequest object and calls
+// QueryValues. The response is parsed by a goroutine and written to the
+// returned channel.
+func (c *client) Query(ctx context.Context, id *api.Identifier) (<-chan *api.ValueList, error) {
+	stream, err := c.client.QueryValues(ctx, &pb.QueryValuesRequest{
+		Identifier: MarshalIdentifier(id),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	ch := make(chan *api.ValueList)
+
+	go func() {
+		defer close(ch)
+
+		for {
+			res, err := stream.Recv()
+			if err == io.EOF {
+				break
+			}
+			if err != nil {
+				log.Printf("error while receiving value lists: %v", err)
```

Comment:
It would be nice to pass this error back to the caller (same below). This would also be easier if the goroutine and synchronization would be handled in one place in the code.